### PR TITLE
Clarify naming for MajorityVoting and subclasses

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description
This PR clarifies names to 
- achieve consistency between core and SDK and 
- understandable, descriptive parameter names that can be shown on the frontend

No logic was changed compared to the base branch.

Task: [APP-1250](https://aragonassociation.atlassian.net/browse/APP-1250)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by Rekard0 — DISMISSED** [review]
apart from AddresslistVoting.sol appears as new file, LGTM!

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol [hunk 1]
```diff
@@ -203,7 +203,7 @@ contract ERC20VotingSetup is PluginSetup {
             plugin,
             _dao,
             NO_ORACLE,
-            erc20VotingBase.SET_CONFIGURATION_PERMISSION_ID()
+            erc20VotingBase.CHANGE_VOTE_SETTINGS_PERMISSION_ID()
```
Comment (josemarinas) [inline packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol:206]: `CHANGE_PROPOSAL_SETTINGS_PERMISSION_ID`

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol [hunk 2]
```diff
@@ -274,7 +274,7 @@ contract ERC20VotingSetup is PluginSetup {
             _plugin,
             _dao,
             NO_ORACLE,
-            erc20VotingBase.SET_CONFIGURATION_PERMISSION_ID()
+            erc20VotingBase.CHANGE_VOTE_SETTINGS_PERMISSION_ID()
```
Comment (josemarinas) [inline packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol:277]: Same here

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol (hunk 1 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol:206]: Ah, absolutely. Good catch

File: packages/contracts/contracts/voting/majority/IMajorityVoting.sol [hunk 3]
```diff
@@ -17,126 +17,132 @@ interface IMajorityVoting {
...
         uint64 minDuration
     );
 
-    /// @notice Sets the vote configuration.
+    /// @notice Changes the vote settings.
     /// @param _totalSupportThresholdPct The total support threshold in percent.
     /// @param _relativeSupportThresholdPct The relative support threshold in percent.
     /// @param _minDuration The minimal duration of a vote.
-    function setConfiguration(
+    function changeVoteSettings(
```
Comment (josemarinas) [inline packages/contracts/contracts/voting/majority/IMajorityVoting.sol:72]: Another one here, if we are changing the name from Vote to Proposal this should be `changeProposalSettings`

Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/majority/IMajorityVoting.sol:72]: On second thought: these are really the vote settings.
In the next step, we will probably make a struct

`struct VoteSettings`
containing the parameters that is different from `struct Proposal`.

`MajorityVoting` can contain `VoteSettings` even if we call it a proposal.

File: packages/subgraph/src/packages/addresslist/addresslist-voting.ts [hunk 4]
```diff
@@ -2,11 +2,11 @@ import {BigInt, dataSource, store} from '@graphprotocol/graph-ts';
...
 import {
   VoteCast,
-  VoteCreated,
-  VoteExecuted,
-  ConfigUpdated,
-  UsersAdded,
-  UsersRemoved,
+  ProposalCreated,
+  ProposalExecuted,
+  VoteSettingsUpdated,
```
Comment (josemarinas) [inline packages/subgraph/src/packages/addresslist/addresslist-voting.ts:7]: same

**Comment by josemarinas** [comment]
Also, as a minor difference with the sdk, on the sdk the address list voting is represented like `addressList` and here is represented as `addresslist`, I don't have a strong opinion on which one is the correct one, but I believe we should agree on one, let me know what you think

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol [hunk 5]
```diff
@@ -21,7 +21,7 @@ import {MerkleDistributor} from "../../tokens/MerkleDistributor.sol";
 import {IERC20MintableUpgradeable} from "../../tokens/IERC20MintableUpgradeable.sol";
 import {ERC20Voting} from "./ERC20Voting.sol";
 
-/// @title AllowlistVotingSetup
+/// @title AddresslistVotingSetup
```
Comment (mathewmeconry) [inline packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol:24]: ERC20VotingSetup

File: active_contracts.json [hunk 6]
```diff
@@ -18,7 +18,7 @@
     "DAOFactory": "0xd52CA6BbfC11627aF9aC3496CAb975ae20E70A81",
     "Registry": "0x2B8C4DD137104d1E869105cd0106e7D9EF955BfE",
     "TokenFactory": "0xE0b582144d9347e849Ab91780f904d4002c55E05",
-    "AllowlistVotingSetup": "0xD39cE95D1e29d4EA9F4E8813d73E68F2F720377a",
+    "AddresslistVotingSetup": "0xD39cE95D1e29d4EA9F4E8813d73E68F2F720377a",
```
Comment (mathewmeconry) [inline active_contracts.json:21]: Please don't rename it here. the address deployed is stil AllowlistVotingSetup and not AddresslistVotingSetup

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol (hunk 5 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol]: Done

File: active_contracts.json (hunk 6 above)
Comment (Michael-A-Heuer) [reply active_contracts.json]: Done

**Comment by Michael-A-Heuer** [comment]
> Also, as a minor difference with the sdk, on the sdk the address list voting is represented like `addressList` and here is represented as `addresslist`, I don't have a strong opinion on which one is the correct one, but I believe we should agree on one, let me know what you think

As there seems to be missing alignment on the naming: https://discord.com/channels/672466989217873929/950422827893420153/1045348071942410280, I suggest we leave the contract names as they are for now.

**Comment by Michael-A-Heuer** [comment]
> > Also, as a minor difference with the sdk, on the sdk the address list voting is represented like `addressList` and here is represented as `addresslist`, I don't have a strong opinion on which one is the correct one, but I believe we should agree on one, let me know what you think
> 
> As there seems to be missing alignment on the naming: https://discord.com/channels/672466989217873929/950422827893420153/1045348071942410280, I suggest we leave the contract names as they are for now.

I created a JIRA task so that we can merge this PR and unblock me for now: https://aragonassociation.atlassian.net/browse/APP-1321

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol (hunk 1 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol:206]: We tacke this in task https://aragonassociation.atlassian.net/browse/APP-1321

File: packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/erc20/ERC20VotingSetup.sol:277]: see https://github.com/aragon/core/pull/146#discussion_r1031554135

File: packages/contracts/contracts/voting/majority/IMajorityVoting.sol (hunk 3 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/voting/majority/IMajorityVoting.sol:72]: see https://github.com/aragon/core/pull/146#discussion_r1031554135

**Review by josemarinas — APPROVED** [review]

**Review by mathewmeconry — APPROVED** [review]

**Pull request merged**